docs/template_plugin/tests/functional/op_reference/acos.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <ie_core.hpp>
+#include <ie_ngraph_utils.hpp>
+#include <ngraph/ngraph.hpp>
+#include <shared_test_classes/base/layer_test_utils.hpp>
+#include <vector>
+
+#include "base_reference_test.hpp"
+
+using namespace ngraph;
+
+namespace reference_tests {
+namespace {
+
+struct AcosParams {
+    Tensor input;
+    Tensor expected;
+};
+
+struct Builder : ParamsBuilder<AcosParams> {
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, input);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, expected);
+};
+
+class ReferenceAcosLayerTest : public testing::TestWithParam<AcosParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params.input.shape, params.input.type);
+        inputData = {params.input.data};
+        refOutData = {params.expected.data};
+    }
+    static std::string getTestCaseName(const testing::TestParamInfo<AcosParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "shape=" << param.input.shape << "_";
+        result << "type=" << param.input.type;
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const Shape& shape, const element::Type& type) {
+        const auto in = std::make_shared<op::Parameter>(type, shape);
+        const auto acos = std::make_shared<op::Acos>(in);
+        return std::make_shared<Function>(NodeVector {acos}, ParameterVector {in});
+    }
+};
+
+TEST_P(ReferenceAcosLayerTest, AcosWithHardcodedRefs) {
+    Exec();
+}
+
+}  // namespace
+
+INSTANTIATE_TEST_SUITE_P(
+    smoke_Acos_With_Hardcoded_Refs, ReferenceAcosLayerTest,
+    ::testing::Values(Builder {}
+                          .input({{11}, element::f16, std::vector<ngraph::float16> {-1.f, -0.75f, -0.5f, -0.25f, -0.125f,
```

Comment:
Could you cover all supported data types?

Reply:
input vector for int will have only {-1, 0, 1}, so only 3 elements are ok for test?

Reply:
Add cases for int32, int64, uint32, uint64.

Reply:
As I see acos evaluate supports next data types:
```
    case ngraph::element::i32:
    case ngraph::element::i64:
    case ngraph::element::u32:
    case ngraph::element::u64:
    case ngraph::element::f16:
    case ngraph::element::f32:
    case ngraph::element::boolean:
```
Can you cover all supported data types?

Reply:
I can see support data type includes boolean type in ov::op::v0::Acos::has_evaluate(). But couldn't find any test case of boolean type in reference test cases of other trigonometric function operations. Is it enough to use {false, true} input as {0, 1}?

Reply:
When I tried element::boolean in the op reference test, error happened in validate_and_infer_elementwise_arithmetic(). It doesn't allow the element to have boolean when is_dynamic()=false.

https://github.com/wilson-seok/openvino/blob/f347f83b0a02adb47a1ffdd38c81ed2b393b8773/ngraph/core/src/op/util/unary_elementwise_arithmetic.cpp#L16

Reply:
@ilyachur Could you please confirm that we can skip boolean type test in the reference test?

Reply:
I think that we should remove boolean from the evaluate method.

Reply:
Ok, I removed it from evaluate_acos().